pontoon/base/management/commands/update_auth_providers.py

```diff
@@ -0,0 +1,75 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# This file has been copied from http://github.com/mozilla/testpilot
+# and has been modified in order to use Fxa provider from the allauth package.
+from __future__ import absolute_import
+
+from django.core.exceptions import ObjectDoesNotExist
+from django.core.management.base import BaseCommand
+from django.conf import settings
+from django.contrib.sites.models import Site
+from allauth.socialaccount.models import SocialApp
+from allauth.socialaccount.providers.fxa.provider import FirefoxAccountsProvider
+from allauth.socialaccount.providers.github.provider import GithubProvider
+
+
+FXA_PROVIDER_ID = FirefoxAccountsProvider.id
+GITHUB_PROVIDER_ID = GithubProvider.id
+
+
+class Command(BaseCommand):
+    help = ('Ensures an allauth application exists and has credentials that match settings')
+
+    def handle(self, *args, **options):
+        def update_provider(data):
```

Comment:
Can you make this a standalone function, not part of `handle()`? You'll need to call it with `this.update_provider()`.

Reply:
Moving this out of `handle()` causes "undefined name 'self' " errors on the 2 write lines in `update_provider`  
1)`self.stdout.write("Updating ....`
2) `self.stdout.write("Created ....`

Removing `self` generates an error of `undefined name 'stdout'`

Reply:
The first argument of update_provider must be self I guess:
```python
def update_provider(self, data):
```